dev-tools/packaging/packages.yml

```diff
@@ -584,6 +584,115 @@ shared:
         source: '{{ repo.RootDir }}/dev-tools/licenses/ELASTIC-LICENSE-2.0.txt'
         mode: 0644
 
+platforms: &all-platforms
+  - &linux-amd64
+    os: linux
+    arch: x86_64
+  - &linux-arm64
+    os: linux
+    arch: arm64
+  - &windows-amd64
+    os: windows
+    arch: x86_64
+  - &darwin-amd64
+    os: darwin
+    arch: x86_64
+  - &darwin-arm
```

Comment:
I thought we also produced a `darwin-universal` package, but it looks like Agent does not produce one.  We produce one for Endpoint for DRA.  Looks like we can possibly get rid of that one at some point if Agent isn't planning on ever producing one.

Reply:
Elastic Agent doesn't produce a`darwin-universal` package.
In this PR I just moved the dependency listing from go code to YAML without changing it.
I also don't think this dependency is ever used but I would prefer if we perform a cleanup in a separate change, to reduce risk :sweat_smile: